src/a_star_search/core.clj

```diff
@@ -1,32 +1,34 @@
 (ns a-star-search.core
   (:require [search.algorithms :as alg]
-            [problems.n-puzzle :as np])
+            [problems.hanoi :as np])
   (:gen-class))
 
 (defn print-results [result path costs goal-state]
   (println (str "We explored " (count result) " states."))
   (println "The path to the solution is:")
-  (doseq [b (map :board path)]
+  (doseq [b (map :towers path)]
     (println b))
   (when costs
     (println "The path has" (count path) "steps and its cost is" (costs goal-state))))
 
 (defn -main
-  "Search for a hard-coded N-puzzle target state."
+  "Search for a hard-coded Towers of Hanoi target state."
   [& args]
-  (let [start-state (np/->State [[0 1 3] [4 2 5] [7 8 6]] [0 0])
-        goal-state (np/->State [[0 1 2] [3 4 5] [6 7 8]] [0 0])
+  ;; Need to change np/ -> State that will work with our code
+  (let [start-state (np/->State [[5 4 3 2 1] [] []])
+        goal-state (np/->State [[] [] [5 4 3 2 1]])
         max-states 1000000
         costs nil
+        ;; here too, at np/children, to work for with our code.
         came-from (alg/breadth-first-search np/children max-states start-state goal-state)
-        ; [came-from costs] (alg/shortest-path np/children (constantly 1)
-        ;                                      max-states start-state goal-state )
-        ; [came-from costs] (alg/shortest-path np/children np/prefer-horizontal-cost
-        ;                                      max-states start-state goal-state)
+        [came-from costs] (alg/shortest-path np/children (constantly 1)
```

Comment:
You only want _one_ of the search calls to be active at a time, and you've got all three going (breadth-first, shortest path with a constant cost, and shortest path with the `prefer-horizontal-cost` (which doesn't even make sense in your puzzle).